mods/ra/maps/ant-03/ant-03.lua

```diff
@@ -0,0 +1,199 @@
+			Trigger.RemoveProximityTrigger(id)
+			Hive2Gassed = true
+			Actor.Create("flare", true, { Owner = England, Location = Hive2.Location })
+			Actor.Create("flare", true, { Owner = England, Location = Hive2Flare.Location })
+			Trigger.OnEnteredFootprint(Hive2Killzone, function(ant)
+				if ant.Type == "warriorant" then
+					ant.Kill("ExplosionDeath")
+				end
+			end)
+		end
+	end)
+
+	Trigger.OnEnteredProximityTrigger(Hive3.CenterPosition, WDist.FromCells(1), function(actor, id)
+		if actor.Type == "chan" then
+			Trigger.RemoveProximityTrigger(id)
+			Hive3Gassed = true
+			Actor.Create("flare", true, { Owner = England, Location = Hive3.Location })
+			Actor.Create("flare", true, { Owner = England, Location = Hive3Flare.Location })
+			Trigger.OnEnteredFootprint(Hive3Killzone, function(ant)
+				if ant.Type == "warriorant" then
+					ant.Kill("ExplosionDeath")
+				end
+			end)
+		end
+	end)
+
+	Trigger.OnEnteredProximityTrigger(Hive4.CenterPosition, WDist.FromCells(1), function(actor, id)
+		if actor.Type == "chan" then
+			Trigger.RemoveProximityTrigger(id)
+			Hive4Gassed = true
+			Actor.Create("flare", true, { Owner = England, Location = Hive4.Location })
+			Actor.Create("flare", true, { Owner = England, Location = Hive4Flare.Location })
+			Trigger.OnEnteredFootprint(Hive4Killzone, function(ant)
+				if ant.Type == "warriorant" then
+					ant.Kill("ExplosionDeath")
+				end
+			end)
+		end
+	end)
+
+	Trigger.OnEnteredProximityTrigger(Hive5.CenterPosition, WDist.FromCells(1), function(actor, id)
+		if actor.Type == "chan" then
+			Trigger.RemoveProximityTrigger(id)
+			Hive5Gassed = true
+			Actor.Create("flare", true, { Owner = England, Location = Hive5.Location })
+			Actor.Create("flare", true, { Owner = England, Location = Hive5Flare.Location })
+			Trigger.OnEnteredFootprint(Hive5Killzone, function(ant)
+				if ant.Type == "warriorant" then
+					ant.Kill("ExplosionDeath")
+				end
+			end)
+		end
+	end)
+
+	Trigger.OnEnteredProximityTrigger(Hive6.CenterPosition, WDist.FromCells(1), function(actor, id)
+		if actor.Type == "chan" then
+			Trigger.RemoveProximityTrigger(id)
+			Hive6Gassed = true
+			Actor.Create("flare", true, { Owner = England, Location = Hive6.Location })
+			Actor.Create("flare", true, { Owner = England, Location = Hive6Flare.Location })
+			Trigger.OnEnteredFootprint(Hive6Killzone, function(ant)
+				if ant.Type == "warriorant" then
+					ant.Kill("ExplosionDeath")
+				end
+			end)
+		end
+	end)
+
+	Trigger.OnEnteredProximityTrigger(Hive7.CenterPosition, WDist.FromCells(1), function(actor, id)
+		if actor.Type == "chan" then
+			Trigger.RemoveProximityTrigger(id)
+			Hive7Gassed = true
+			Actor.Create("flare", true, { Owner = England, Location = Hive7.Location })
+			Actor.Create("flare", true, { Owner = England, Location = Hive7Flare.Location })
+			Trigger.OnEnteredFootprint(Hive7Killzone, function(ant)
+				if ant.Type == "warriorant" then
+					ant.Kill("ExplosionDeath")
+				end
+			end)
+		end
+	end)
```

Comment:
There's quite a lot of repeated code here. I think you could use arrays to reduce it to a loop:
```
for nestId = 1, 7 do
	Trigger.OnEnteredProximityTrigger(Hives[nestId].CenterPosition, WDist.FromCells(1), function(actor, id)
		if actor.Type == "chan" then
			Trigger.RemoveProximityTrigger(id)
			HiveGassed[nestId] = true
			Actor.Create("flare", true, { Owner = England, Location = Hives[nestId].Location })
			Actor.Create("flare", true, { Owner = England, Location = HiveFlares[nestId].Location })
			Trigger.OnEnteredFootprint(HiveKillzones[nestId], function(ant)
				if ant.Type == "warriorant" then
					ant.Kill("ExplosionDeath")
				end
			end)
		end
	end)
end
```